connector-agent-python/src/lib.rs

```diff
@@ -72,3 +72,40 @@ fn read_pg(py: Python, conn: &str, sqls: Vec<String>, schema: &str) -> PyResult<
         .collect();
     PyResult::Ok(ret.into_py_dict(py).to_object(py))
 }
+
+#[pyfunction]
+fn read_sql<'a>(
+    py: Python<'a>,
+    conn: &str,
+    query: &str,
+    col: &str,
```

Comment:
Can we make the partition optional? i.e. pass in a tuple or just first create a struct called `PartitionConfig` and derive IntoPyDict for it. for `read_sql` it takes py, conn, query and Option<PartitionConfig> and return_type. If the partition is not specified, we do not do partition.

Reply:
Also I feel like we need to have a wrapper on the python side to bundle the params and make some of them keyword only to make the intention of the function clear.